src/App.jsx

```diff
@@ -14,33 +15,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortState, setSortState] = useState({
+    reversed: false,
+    lengthSorted: false,
+    alphabetSorted: false,
+  });
+  const [isReset, setIsReset] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortGoodsByAlphabet = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.localeCompare(a);
+      }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+      return a.localeCompare(b);
+    });
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, alphabetSorted: true, lengthSorted: false });
+    setIsReset(true);
+  };
+
+  const sortGoodsByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.length - a.length;
+      }
+
+      return a.length - b.length;
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, lengthSorted: true, alphabetSorted: false });
+    setIsReset(true);
+  };
+
+  const reverseGoods = () => {
+    setSortState(prevState => {
+      const reversed = !prevState.reversed;
+
+      setGoods(prevGoods => {
+        const newGoods = [...prevGoods].reverse();
+
+        setIsReset(
+          !newGoods.every(
+            (element, index) => element === goodsFromServer[index],
+          ),
+        );
+
+        return newGoods;
+      });
```

Comment:
The `reverseGoods` function is a bit complex due to the nested `setGoods` call. Consider simplifying this by reversing the goods list outside of the `setSortState` function and then updating the state. This will make the code more readable and maintainable.